src/Tools/AttachEntityListenersListener.php

```diff
@@ -5,25 +5,33 @@
 namespace Doctrine\ORM\Tools;
 
 use Doctrine\ORM\Event\LoadClassMetadataEventArgs;
+use Doctrine\ORM\Events;
 use Doctrine\ORM\Mapping\Builder\EntityListenerBuilder;
 
+use function assert;
 use function ltrim;
 
 /**
  * Mechanism to programmatically attach entity listeners.
  */
 class AttachEntityListenersListener
 {
-    /** @var mixed[][] */
+    /**
+     * @var array<class-string, list<array{
+     *     event: Events::*|null,
+     *     class: class-string,
+     *     method: string|null,
+     * }>>
+     */
     private array $entityListeners = [];
 
     /**
      * Adds an entity listener for a specific entity.
      *
-     * @param string      $entityClass      The entity to attach the listener.
-     * @param string      $listenerClass    The listener class.
-     * @param string|null $eventName        The entity lifecycle event.
-     * @param string|null $listenerCallback The listener callback method or NULL to use $eventName.
+     * @param class-string          $entityClass      The entity to attach the listener.
+     * @param class-string          $listenerClass    The listener class.
+     * @param Events::*|null        $eventName        The entity lifecycle event.
+     * @param non-falsy-string|null $listenerCallback The listener callback method or NULL to use $eventName.
      */
     public function addEntityListener(
```

Comment:
Note: it's unclear to me who/what calls this method. The class is not referenced at all in `src`, but it is referenced in the Symfony bundle.

Reply:
Possible. I remember when entitylisteners could only be configured with annotations in the entity class itself and there was no way to inject dependencies into the listener's constructor.